include/zephyr/drivers/stepper_motor_controller.h

```diff
@@ -0,0 +1,279 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_STEPPER_MOTOR_CONTROLLER_H
+#define ZEPHYR_INCLUDE_STEPPER_MOTOR_CONTROLLER_H
+
+/** \addtogroup stepper_motor_controller
+ *  @{
+ */
+
+/**
+ * @file
+ * @brief Public API for Stepper Motor Controller
+ *
+ */
+
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Stepper Motor Position Format required while setting or getting stepper motor positions.
+ */
+enum micro_step_resolution {
+	FULL_STEP,
+	MICRO_STEP_2,
+	MICRO_STEP_4,
+	MICRO_STEP_8,
+	MICRO_STEP_16,
+	MICRO_STEP_32,
+	MICRO_STEP_64,
+	MICRO_STEP_128,
+	MICRO_STEP_256,
+};
+
+enum direction {
+	POSITIVE,
+	NEGATIVE,
+};
+
+/**
+ * @typedef stepper_motor_controller_enable_t
+ * @brief enable or disable the stepper motor controller.
+ */
+typedef int32_t (*stepper_motor_controller_enable_t)(const struct device *dev, bool enable);
+
+/**
+ * @typedef stepper_motor_controller_set_steps_to_move_t
+ * @brief Stepper Motor Controller shall move the motor with given steps
+ */
+typedef int32_t (*stepper_motor_controller_set_steps_to_move_t)(const struct device *dev,
```

Comment:
**suggestion:** IMHO micro steps are meant to smoothen the motor motion not necessarely to sub-divide achievable position resolution, so when talking about steps it probably makes more sense to almost always refer to full-steps